### PR TITLE
Restructure the CIS and DISA STIG hardening guides

### PR DESCRIPTION
**Pull request by ktsakalozos**
- rename the CIS and DISA STIG hardening pages to assessments
- include the DISA STIG assessment report
- have the post-deployment hardening steps in one place
- CIS assessment page includes all the checks

### CODE REVIEW

**Review by bschimke95 — APPROVED** [review]
LGTM

**Comment by bschimke95** [comment]
most spell-checker errors seem unrelated but "unsecure" should probably be "insecure"
https://github.com/canonical/k8s-snap/actions/runs/12275066803/job/34249339507?pr=890#step:4:27

File: docs/src/snap/howto/security/cis-assessment.md [hunk 1]
```diff
@@ -1,31 +1,143 @@
...
+Out of the box {{product}} complies with the majority of the recommended
+CIS security configurations. Since implementing all security recommendations
 would comes at the expense of compatibility and/or performance we expect
 cluster administrators to follow post deployment hardening steps based on their
-needs.
+needs. This guide covers:
+
+  * Post-deployment hardening steps you could consider for your {{product}}
+  * Using [kube-bench][] to automatically check whether your Kubernetes
+    clusters are configured according to the [CIS Kubernetes Benchmark][]
```
Comment (aznashwan) [inline docs/src/snap/howto/security/cis-assessment.md:13]: Missing URL to: cisecurity.org/benchmark/kubernetes

File: docs/src/snap/howto/security/cis-assessment.md [hunk 2]
```diff
@@ -1,31 +1,143 @@
...
+By completing these steps, you can ensure your cluster achieves does not fail
+any of the CIS hardening recommendations.
 
 ```{include} ../../../_parts/common_hardening.md
 ```
 
-If you would also like to apply further DISA STIG recommendations please see [additional DISA STIG-specific steps].
+## Assess CIS hardening with kube-bench
+
+Download the latest [kube-bench release][] on your Kubernetes nodes. Make sure
```
Comment (aznashwan) [inline docs/src/snap/howto/security/cis-assessment.md:37]: Redundant `[]`

File: docs/src/snap/howto/security/disa-stig-assessment.md [hunk 3]
```diff
@@ -0,0 +1,6187 @@
...
+reasons for this include: a check on a Kubernetes feature that was removed
+prior to {{product}}'s first release, a check for a component that {{product}}
+does not package, etc.
+
+**Manual**: These checks require manual intervention from a cluster
+administrator, so they cannot be automated.
+
+
+### [V-242381]: The Kubernetes Controller Manager must create unique service accounts for each work payload
+
```
Comment (aznashwan) [inline docs/src/snap/howto/security/disa-stig-assessment.md:43]: I don't think we'll have any good way around this lint check if we insist on having the Finding's original `title` field in the section title here.

I'll look if there's a way to use `<!-- ignore -->` comments or something and add them to the report generation if I do.

File: docs/src/snap/howto/security/disa-stig-assessment.md [hunk 4]
```diff
@@ -0,0 +1,6187 @@
...
+Note: this Finding allows for this argument to be UNSET as well.
+
+```bash
+grep -E -q  '\-\-read-only-port=(0)' '/var/snap/k8s/common/args/kubelet' || echo UNSET
+test $? -eq 0 && echo PASS || echo FAIL
+```
+
+In the default configuration of the `k8s-snap`, resulting output lines will start with `PASS`.
+
```
Comment (aznashwan) [inline docs/src/snap/howto/security/disa-stig-assessment.md:231]: I'll try to fix this directly from the report generator.

File: docs/src/snap/howto/security/disa-stig-assessment.md [hunk 5]
```diff
@@ -0,0 +1,6187 @@
...
+> this address bypass authentication and authorization checks. If this insecure
+> bind address is set to localhost, anyone who gains access to the host on
+> which the Control Plane is running can bypass all authorization and
+> authentication mechanisms put in place and have full control over the entire
+> cluster.
+> 
+> Close or set the insecure bind address by setting the API server's
+> "--insecure-bind-address" flag to an IP or leave it unset and ensure that the
+> "--insecure-bind-port" is not set.
+
```
Comment (aznashwan) [inline docs/src/snap/howto/security/disa-stig-assessment.md:256]: I'll try to reduce the number of irrelevant lines in the generation of the final report.

File: docs/src/snap/howto/security/hardening.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## CIS and DISA STIG Hardening 
+
+If you would like to apply further DISA STIG recommendations please see [additional DISA STIG-specific steps].
+
+To asses compliance to the CIS hardening guidelines please see the [CIS assessment page].
+
+<!-- Links -->
+[Post-Deployment Configuration Steps section]:#post-deployment-configuration-steps
```
Comment (aznashwan) [inline docs/src/snap/howto/security/hardening.md:34]: I don't see this referenced anywhere?

Even if it were, shouldn't it have been `#post-deployment-harnening-steps`?

File: docs/src/snap/howto/security/index.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ harden their clusters in accordance with DISA STIG and CIS recommendations.
 :glob:
 :titlesonly:
 
-CIS Hardening <cis-hardening.md>
-DISA STIG Hardening <disa-stig-hardening.md>
-Hardening guide <auditing-steps.md>
+Post deployment hardening guide <hardening-post.md>
```
Comment (aznashwan) [inline docs/src/snap/howto/security/index.md:15]: I'm not sure where this is supposed to point?

There's no `hardening-post.md` anywhere (yet...)

**Review by aznashwan — CHANGES_REQUESTED** [review]
Minor comments; most are to myself and I'll address them in the DISA checker tool ASAP.

@ktsakalozos note that a couple of the footer links are either obsolete/not pointing to pages that exist [yet]

File: docs/src/snap/howto/security/index.md (hunk 7 above)
Comment (eaudetcobello) [reply docs/src/snap/howto/security/index.md:15]: It's for hardening.md

File: docs/src/snap/howto/security/hardening.md (hunk 6 above)
Comment (eaudetcobello) [reply docs/src/snap/howto/security/hardening.md:34]: Removed

**Review by nhennigan — APPROVED** [review]

**Pull request merged**